sdk/core/azure-core/src/main/java/com/azure/core/http/rest/ResponseConstructorsCache.java

```diff
@@ -71,40 +167,77 @@
         return null;
     }
 
+    private static Object getModule(Class<?> clazz, ClientLogger logger) {
+        try {
+            return GET_MODULE.invoke(clazz);
+        } catch (Throwable throwable) {
+            throw logger.logExceptionAsError(Exceptions.propagate(throwable));
+        }
+    }
+
+    private static boolean isModuleExported(Object module, ClientLogger logger) {
+        try {
+            return (boolean) IS_MODULE_EXPORTED.invoke(module, "");
+        } catch (Throwable throwable) {
+            throw logger.logExceptionAsError(Exceptions.propagate(throwable));
+        }
+    }
+
+    private static boolean canReadModule(Object module, ClientLogger logger) {
+        try {
+            return (boolean) CAN_READ_MODULE.invoke(CORE_MODULE, module);
+        } catch (Throwable throwable) {
+            throw logger.logExceptionAsError(Exceptions.propagate(throwable));
+        }
+    }
+
+    private static void addModuleRead(Object module, ClientLogger logger) {
+        try {
+            ADD_MODULE_READ.invoke(CORE_MODULE, module);
+        } catch (Throwable throwable) {
+            throw logger.logExceptionAsError(Exceptions.propagate(throwable));
+        }
+    }
```

Comment:
For these methods, catch/log and throw may not be required. The static block that calls these methods, ignores these exceptions and just logs a verbose message with the exception. So, logging them here as an error can be misleading. Calls from `locateResponseConstructor` method should handle the exception and fallback to default lookup.